pd/src/info/specific.rs

```diff
@@ -98,15 +98,23 @@ impl SpecificQuery for Info {
         let state = self.state_tonic().await?;
         state.check_chain_id(&request.get_ref().chain_id).await?;
 
-        let req_inner = request.into_inner();
-        let req_proof = req_inner.proof;
-        let req_key = req_inner.key;
+        let request = request.into_inner();
+        tracing::debug!(?request);
+
+        if request.proof {
+            if request.key.is_empty() {
+                return Err(Status::invalid_argument("key is empty"));
+            }
+            if !request.key_hash.is_empty() {
+                return Err(Status::invalid_argument(
+                    "key_hash is nonempty but proof was requested",
```

Comment:
why do we require key_hash when a proof is requested? couldn't we just derive it from key?

Reply:
we require that `key_hash` is unset, because we can only generate proofs with a `key`, and we don't want to accept both a key and a hash. definitely ugly